src/.github/workflows/pre-commit.yml.jinja

```diff
@@ -13,5 +13,8 @@ jobs:
         with:
           # The pylint-odoo version we use here does not support python 3.10
           # https://github.com/OCA/oca-addons-repo-template/issues/80
-          python-version: "3.9"
+          # We also need to pin to an older version of python for older odoo versions
+          # where we are not using black > 21. Older black versions won't work with
+          # Python 3.9.8, and we can't bump black without reformatting.
+          python-version: "3.9{% if odoo_version < 15 %}.7{% endif %}"
```

Comment:
Why putting conditions and not directly `3.9.7` for all?

Reply:
Yes I think we can have 3.9.7 for all < 15, and 3.10 for odoo_version >= 15.

Reply:
I would pin this preventively. We don't need latest Python version for such simple CI. It's not the test one.

Reply:
Ok, I will do as @sbidoul suggests then... But won't https://github.com/OCA/oca-addons-repo-template/issues/80 affect us for v15 branches?

Reply:
I have tested pylint-odoo 5.0.4 with python 3.10 and it works fine.
An alternative is upgrading black on branches 14 but that will trigger some minor code layout changes, so better not do that now.

Reply:
Ok, I already pushed the change with `3.9.7` for branches bellow v15, and `3.10` for v15.